src/package/mod.rs

```diff
@@ -1,55 +1,137 @@
-use crate::args::{BuildSystem, InitArgs, Platform, TargetLanguage};
-use crate::util::{analyzer, copy_recursively};
+pub mod lock;
+pub mod management;
+pub mod tree;
 
-use serde_derive::{Deserialize, Serialize};
+pub mod target_properties;
 
+use git2::Repository;
+use serde::de::{Error, Visitor};
+use serde::{Deserializer, Serializer};
+use serde_derive::{Deserialize, Serialize};
 use std::collections::HashMap;
+use tempfile::tempdir;
+use versions::Versioning;
+use which::which;
 
 use std::fs::{read_to_string, remove_dir_all, remove_file, write};
 use std::io::ErrorKind;
 use std::path::{Path, PathBuf};
-use std::{env, io};
-
-use crate::args::BuildSystem::{CMake, LFC};
-use crate::util::errors::{BuildResult, LingoError};
-use git2::Repository;
-use tempfile::tempdir;
-use which::which;
+use std::str::FromStr;
+use std::{env, fmt, io};
+
+use crate::args::{
+    BuildSystem,
+    BuildSystem::{CMake, LFC},
+    InitArgs, Platform, TargetLanguage,
+};
+use crate::package::{
+    target_properties::{
+        AppTargetProperties, AppTargetPropertiesFile, LibraryTargetProperties,
+        LibraryTargetPropertiesFile,
+    },
+    tree::PackageDetails,
+};
+use crate::util::{
+    analyzer, copy_recursively,
+    errors::{BuildResult, LingoError},
+};
+
+/// place where are the build artifacts will be dropped
+pub const OUTPUT_DIRECTORY: &str = "build";
+/// name of the folder inside the `OUTPUT_DIRECTORY` where libraries
+/// will be loaded (cloned, extracted, copied) into for further processing.
+pub const LIBRARY_DIRECTORY: &str = "libraries";
```

Comment:
Is this the Lingo's equivalent of Node's `node_modules`?

Reply:
If so, it might be good to feature its name and not use a generic one like "libraries."

Reply:
`build` equivalent to `target` from rust. 

the `build/libraries` folder is used to build the whole depndency tree.